### PR TITLE
Communication board

### PR DESCRIPTION
**Pull request by NoahSprenger**
This PR is a starting place for the communication board. This board assumes the task of sending data over the radio. This PR also renames the logic board to sensor board as it is a more accurate name.

### CODE REVIEW

File: boards/communication/src/communication.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+use crate::data_manager::DataManager;
```
Comment (NoahSprenger) [inline boards/communication/src/communication.rs:1]: Move below the file doc.

File: boards/communication/src/data_manager.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                }
+                messages::sensor::SensorData::Imu2(imu2_data) => {
+                    self.imu.1 = Some(imu2_data);
+                }
+                messages::sensor::SensorData::UtcTime(utc_time_data) => {
+                    self.utc_time = Some(utc_time_data);
+                }
+            },
+            _ => {}
+        }
```
Comment (NoahSprenger) [inline boards/communication/src/data_manager.rs:67]: Clippy: you seem to be trying to use `match` for destructuring a single pattern. Consider using `if let`

File: boards/communication/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            },
+            Local { led, radio },
+            init::Monotonics(mono),
+        )
+    }
+
+    /// Idle task for when no other tasks are running.
+    #[idle]
+    fn idle(_: idle::Context) -> ! {
+        loop {}
```
Comment (NoahSprenger) [inline boards/communication/src/main.rs:125]: Clippy doesn't like this. We can always just remove the idle.

File: boards/communication/src/main.rs
Comment (NoahSprenger) [inline boards/communication/src/main.rs:1]: We can add a function that reads the status of the radio if we want to create a message priority system.

File: boards/recovery/src/data_manager.rs [hunk 4]
```diff
@@ -23,29 +21,27 @@ impl DataManager {
...
             current_state: None,
         }
     }
     /// Returns true if the rocket has passed apogee.
     /// This is determined by looking at the historical pressure data.
-    /// Furthermore, we only start checking pressure data when velocity is less than 20m/s 
-    /// because we want to avoid the complexities of pressure during transonic flight. 
+    /// Furthermore, we only start checking pressure data when velocity is less than 20m/s
+    /// because we want to avoid the complexities of pressure during transonic flight.
     pub fn is_falling(&self) -> bool {
```
Comment (NoahSprenger) [inline boards/recovery/src/data_manager.rs:38]: We need to test this in pressure chamber (*pressure cooker)

File: boards/recovery/src/data_manager.rs [hunk 5]
```diff
@@ -55,65 +51,57 @@ impl DataManager {
...
                 }
-            },
+            }
             None => {
                 return false;
             }
         }
         true
     }
     pub fn is_launched(&self) -> bool {
```
Comment (NoahSprenger) [inline boards/recovery/src/data_manager.rs:61]: We should test this, maybe a simple throw it up in the air?

File: boards/recovery/src/data_manager.rs [hunk 6]
```diff
@@ -55,65 +51,57 @@ impl DataManager {
...
     }
     pub fn is_below_main(&self) -> bool {
-        match self.sbg_air.as_ref() {
-            Some(air) => {
-                air.altitude < MAIN_HEIGHT
-            },
-            None => {
-                false
-            }
+        match self.air.as_ref() {
```
Comment (NoahSprenger) [inline boards/recovery/src/data_manager.rs:68]: We need to figure out if this is Above ground or sea level.

File: boards/recovery/src/data_manager.rs [hunk 7]
```diff
@@ -55,65 +51,57 @@ impl DataManager {
...
         match data.data {
             messages::Data::Sensor(sensor) => match sensor.data {
                 messages::sensor::SensorData::Air(air_data) => {
                     let pressure = air_data.pressure_abs;
-                    self.sbg_air = Some(air_data);
+                    self.air = Some(air_data);
                     if self.historical_pressure.is_full() {
                         self.historical_pressure.dequeue();
                     }
                     self.historical_pressure.enqueue(pressure);
```
Comment (NoahSprenger) [inline boards/recovery/src/data_manager.rs:82]: Handle the result!

File: boards/recovery/src/data_manager.rs [hunk 8]
```diff
@@ -55,65 +51,57 @@ impl DataManager {
...
-                },
+                    self.imu.1 = Some(imu2_data);
+                }
                 messages::sensor::SensorData::UtcTime(utc_time_data) => {
-                    self.sbg_utc_time = Some(utc_time_data);
-                },
+                    self.utc_time = Some(utc_time_data);
+                }
             },
             _ => {}
```
Comment (NoahSprenger) [inline boards/recovery/src/data_manager.rs:106]: Clippy: you seem to be trying to use `match` for destructuring a single pattern. Consider using `if let`

File: boards/sensor/src/main.rs [hunk 9]
```diff
@@ -45,7 +44,6 @@ mod app {
     #[local]
     struct Local {
         led: Pin<PA14, PushPullOutput>,
-        radio: RadioDevice,
         sd_manager: SdManager,
```
Comment (NoahSprenger) [inline boards/sensor/src/main.rs:47]: SD logging can be another PR.

**Review by NoahSprenger — COMMENTED** [review]
TESTING TESTING TESTING!!!!!!!

File: boards/communication/src/main.rs
Comment (NoahSprenger) [reply boards/communication/src/main.rs:1]: This will be a separate PR

**Review by Text-Input — APPROVED** [review]
Looks good. A few comments and concerns we went over in person, but seems good.

File: boards/recovery/src/state_machine/states/apogee.rs
Comment (NoahSprenger) [inline boards/recovery/src/state_machine/states/apogee.rs:1]: Super important!
